react-components/src/apps/admin/AllStats/AppStatsComponents/datePicker.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Used for Start Date & End Date
+ */
+
+import React from "react";
+import {
+    MuiPickersUtilsProvider,
+    KeyboardDatePicker,
+} from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
+import { formatDateObject } from "@cyverse-de/ui-lib/src/util/DateFormatter";
+
+function DatePicker(props) {
+    const [selectedDate, setSelectedDate] = React.useState(new Date());
+    const handleDateChange = (date) => {
+        let formattedDate = formatDateObject(date, "YYYY-MM-DD");
+        console.log(formattedDate);
+        setSelectedDate(date);
```

Comment:
One more refactor is required that will allow a date change to trigger an API call, which will reload these components with new data. Just as the story is now passing in mock data for these components to render, the story should also pass down a mock API callback function that can be used when the date changes.

In our current design pattern, we are wrapping API calls in a `presenter` object that is also passed into the view components, just as the data is now passed in.

Take a look at the example in the Teams story:
https://github.com/cyverse-de/ui/blob/5778ad4aa31b03168e65f92fb26dcea18c935a8b/react-components/stories/teams/Teams.stories.js#L14-L23

This mock presenter is defining a few mock callback functions. You can create a simple API callback mock similar to this `onTeamNameSelected` method. Your story's mock callback will replace this `handleDateChange` and it can format and log the date just like this one.

I'm also thinking your story component will also have to manage the state of the selected dates, and the dates will have to be passed down to these components, just as the mock data is currently passed down.

Reply:
Agreed.  Although I'm not 100% sure how different this will look in sonora, this is still a good pattern to follow to create the mental model of that flow between the view and the API.